Automattic/sensei
# Create new course general sidebar

Fixes #6028 

### Changes proposed in this Pull Request

* Adds a 'General' panel to the new Sensei settings sidebar
* Adds a new item for each meta box that is displayed in the course settings
* Maintains the same meta-boxes when classic editor is used
* Adds a rest API route to update course teacher when post is updated

### Testing instructions

- Switch to the classic editor, 
- Edit a course and notice that these meta-boxes are displayed on the sidebar: teacher, course prerequisite, featured course, notifications etc
- Switch back to the block editor, a course.
- Click on Sensei Settings in the sidebar.
- Notice the 'General' panel opened. 
- Check that all the items in the meta-box from the classic editor are in this panel: teacher, course prerequisite, featured course, notifications.


### Known Bugs
Saving the course teacher works fine with the custom rest API. 
However, I noticed a strange bug where updating the post doesn't persist the new values set in the post meta fields using: 
`[ meta, setMeta ] = useEntityProp( 'postType', 'course', 'meta' );`, 
I may need a fresh pair of eyes to have a look.

### Screenshots

<img width="422" alt="Screenshot 2022-11-08 at 14 29 05" src="https://user-images.githubusercontent.com/26160845/200575514-5ef2d892-14d2-468f-a3aa-3730d0f7303c.png">

UPDATE: The issue with saving meta has been fixed.
